packages/orchestration/src/exos/cosmos-orchestration-account.js

```diff
@@ -785,68 +772,93 @@ export const prepareCosmosOrchestrationAccountKit = (
               'ibc',
-              'cosmos',
-              destination.chainId,
               transferChannel.counterPartyPortId,
               transferChannel.counterPartyChannelId,
             ],
+            seq: { status: 'pending' }, // filled in by fillSequenceWatcher.onFulfilled below
           });
           return watch(
-            allVows([result, meta]),
-            this.facets.parseTransferWatcher,
+            resultMeta,
+            this.facets.updateMetaTrafficWatcher,
             transferTraffic,
           );
         },
       },
       parseTransferWatcher: {
-        onFulfilled([transferResp, baseMeta], transferTraffic) {
-          trace('parseTransferWatcher', {
-            submitted: transferResp,
-            baseMeta,
-            transferTraffic,
+        onFulfilled(_1, _2) {
+          throw Fail`obsolete parseTransferWatcher(${_1}, ${_2}); please retry`;
+        },
+      },
+      updateMetaTrafficWatcher: {
+        /**
+         * @param {ResultMeta<any>} param0
+         * @param {MetaTrafficEntry} traffic
+         * @returns {ResultMeta<{ followTraffic: MetaTrafficEntry }>}
+         */
+        onFulfilled({ result, meta }, traffic) {
+          trace('updateMetaTrafficWatcher', {
+            result,
+            meta,
+            traffic,
           });
-          const { sequence } = tryDecodeResponse(
-            transferResp,
-            MsgTransferResponse,
+          const newMeta = {
+            ...meta,
+            traffic: [...(meta.traffic || []), traffic],
+          };
+          const newResult = watch(
+            result,
+            this.facets.fillSequenceWatcher,
+            newMeta,
           );
-          const baseSequence = transferTraffic?.seq;
-          baseSequence == null ||
-            Fail`expected traffic?.seq ${baseSequence} to be nullish`;
+          return harden({ result: newResult, meta: newMeta });
+        },
+      },
+      fillSequenceWatcher: {
+        /**
+         * @param {string} sequenceResp
+         * @param {Record<string, any>} meta
+         */
+        onFulfilled(sequenceResp, meta) {
+          trace('fillSequenceWatcher', {
+            submitted: sequenceResp,
+            meta,
+          });
+
+          const [{ sequence }] = tryDecodeResponses(sequenceResp, [
+            MsgTransferResponse,
+          ]);
+          const lastMetaTraffic = meta.traffic?.at(-1);
+          lastMetaTraffic ||
+            Fail`expected meta.traffic to have at least one entry: ${q(meta)}`;
+          const baseSequence = lastMetaTraffic?.seq;
+          baseSequence?.status === 'pending' ||
+            Fail`expected traffic?.seq ${baseSequence} to be pending`;
           sequence != null ||
-            Fail`expected MsgTransferResponse.sequence ${sequence} to be non-nullish`;
+            Fail`expected sequenceResp.sequence ${sequence} to be non-nullish`;
 
+          const newTransferTraffic = { ...lastMetaTraffic, seq: sequence };
```

Comment:
above "traffic" refers to a list of these objects and here's it's one. Please use a different term for singular and plural. "Traffic" generally implies many (cars). So I'd suggest `lastMetaTransit` and `newTransit` here.

Reply:
I'm using `Traffic` and `TrafficEntry` now.